sklearn/ensemble/_voting.py

```diff
@@ -153,7 +154,17 @@ def _sk_visual_block_(self):
         return _VisualBlock("parallel", estimators, names=names)
 
     def _more_tags(self):
-        return {"preserves_dtype": []}
+        try:
+            allow_nan = all(
+                _safe_tags(est[1])["allow_nan"] if est[1] != "drop" else True
+                for est in self.estimators
+            )
+        except Exception:
+            # If `estimators` does not comply with our API (list of tuples) then it will
+            # fail. In this case, we assume that `allow_nan` is False but the parameter
+            # validation will raise an error during `fit`.
+            allow_nan = False
```

Comment:
I think this code is being repeated so maybe we could extract it into a separate function either in some utils or common base class?

Reply:
Since the code is checking `est[1]` I would not put it in a util but we can factorize it in the base file of the `ensemble module.

Reply:
Apparently, it integrates well with the base class.